synapse/cortex.py

```diff
@@ -3573,6 +3594,64 @@ async def _delTagProp(self, name):
         await self.fire('core:tagprop:change', name=name, act='del')
         await self.feedBeholder('model:tagprop:del', {'tagprop': name})
 
+    async def addEdge(self, edge, edgeinfo):
+        if not isinstance(edgeinfo, dict):
+            mesg = 'Edge info should be a dict.'
+            raise s_exc.BadArg(mesg=mesg, edgeinfo=edgeinfo)
+
+        (n1form, verb, n2form) = edge
+        if not verb.startswith('_'):
+            mesg = f'Extended edge verb must begin with "_"; got {verb}'
+            raise s_exc.BadEdgeDef(mesg=mesg, n1form=n1form, verb=verb, n2form=n2form)
+
+        if n1form is not None:
+            self.model._reqFormName(n1form)
+
+        if n2form is not None:
+            self.model._reqFormName(n2form)
+
+        if self.model.edge(edge) is not None:
+            raise s_exc.DupEdgeType.init(edge)
+
+        return await self._push('model:edge:add', edge, edgeinfo)
+
+    @s_nexus.Pusher.onPush('model:edge:add')
+    async def _addEdge(self, edge, edgeinfo):
+        if self.model.edge(edge) is not None:
+            return
+
+        self.model.addEdge(edge, edgeinfo)
+
+        (n1form, verb, n2form) = edge
+
+        self.extedges.set(s_common.guid(edge), (edge, edgeinfo))
+        await self.fire('core:extmodel:change', n1form=n1form, verb=verb, n2form=n2form, act='add', type='edge')
+
+        edgedef = {'n1form': n1form, 'verb': verb, 'n2form': n2form, 'info': edgeinfo}
+        await self.feedBeholder('model:edge:add', edgedef)
+
+    async def delEdge(self, edge):
+        if self.extedges.get(s_common.guid(edge)) is None:
+            raise s_exc.NoSuchEdge.init(edge)
+
+        return await self._push('model:edge:del', edge)
+
+    @s_nexus.Pusher.onPush('model:edge:del')
+    async def _delEdge(self, edge):
+        edgeguid = s_common.guid(edge)
+        if self.extedges.get(edgeguid) is None:
+            return
+
+        self.model.delEdge(edge)
+
+        (n1form, verb, n2form) = edge
+
+        self.extedges.pop(edgeguid, None)
+        await self.fire('core:extmodel:change', n1form=n1form, verb=verb, n2form=n2form, act='del', type='edge')
+
+        edgedef = {'n1form': n1form, 'verb': verb, 'n2form': n2form}
```

Comment:
Could we just send along an `edgetype` tuple here (and above)? I'd like to start trying to make these messages a bit more compact :)